### ISSUE TITLE
[Bug]: Moderation syntax does not work in PLS-SEM

### ISSUE THREAD

**Issue by francischuah** [jasp-stats/jasp-issues#3007]
### JASP Version

0.18.3

### JASP Module

SEM

### What analysis are you seeing the problem on?

PLS-SEM

### What OS are you seeing the problem on?

Windows 11

### Bug Description

![PLS SEM moderation does not work](https://github.com/user-attachments/assets/322751c8-e9d0-4f2d-9e1a-8b98257e556d)
[Izzat data.csv](https://github.com/user-attachments/files/17711319/Izzat.data.csv)

### Expected Behaviour

It should work. Model fit cannot be calculated for non-linear model. But I do not need the model fit.

### Steps to Reproduce

Run the following syntax with the above given data.

#latent variable
Product =~ Product1 + Product2 + Product3 + Product4 + Product5
Price =~ Price1 + Price2 + Price3 + Price4 + Price5
Intention =~ Int1 + Int2 + Int3 + Int4 + Int5

#structural model
Intention ~ Product + Price 
Intention ~ Product.Price

### Log (if any)

This analysis terminated unexpectedly.

Error: The following error occured while computing the model-implied indicator correlation matrix:
`fit()` currently not applicable to nonlinear models.

Stack trace

To receive assistance with this problem, please report the message above at: https://jasp-stats.org/bug-reports

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [X] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by juliuspfadt** [comment]
oh, I totally missed this. @FloSchuberth, do you have an idea why this happens?

**Comment by FloSchuberth** [comment]
@juliuspfadt: Yes because some "statistics" are not calculated if the model contains non-linear terms. An example of this is the fit() function, which returns the model-implied correlation matrix, see https://github.com/FloSchuberth/cSEM/blob/master/R/csem_fit.R
Since, you call this function, you get an error.

**Comment by tomtomme** [comment]
@juliuspfadt 
should this error still be scheduled for 0.95?

Error confirmed with: 

-------- Application Info --------
JASP Version: JASP 0.20
Build Branch: HEAD
Build Date: May 21 2025 09:39:55 (Netherlands)
Last Commit: 50d046a6705095a8d4218132d6a5a2ca58dc91c5

-------- Basic Info --------
Operating System: KDE Flatpak runtime
Product Version: 6.8
Kernel Type: linux
Kernel Version: 6.14.6-2-MANJARO
Architecture: x86_64
Install Path: /app/bin
Platfotm Name: wayland
System Local: de_DE

**Comment by juliuspfadt** [comment]
yes. 0.95.0 should be fine, I think?

**Comment by juliuspfadt** [comment]
sorry, I didn't get the memo, that we had replaced 0.20 with 0.95

**Comment by juliuspfadt** [comment]
coming back around to this. @FloSchuberth would there be a way to get the fit for a model with non-linear terms?
